app/components/filtered-by-book-transactions.js

```diff
@@ -0,0 +1,21 @@
+import Component from '@ember/component';
+import { computed } from '@ember/object';
+export default Component.extend({
+  classNames: ['detailedTransactions'],
+    areIdsEqual: computed('currentTransaction', function() {
+      let currentBookFromTransaction = this.get('currentTransaction.book_id');
+      let currentBook = this.get('book.id');
+      if (currentBookFromTransaction === currentBook ){
+        return true;
+      }
+    }),
+    showCorrectBook: computed('isBookShowing','choosedBook', function(){
+      let isBookShowing = this.get('isBookShowing');
+      let currentBook = this.get('book.id');
+      let choosedBook = this.get('choosedBook');
+      if (isBookShowing && choosedBook == currentBook){
```

Comment:
'choosedBook' is incorrect should be chosenBook. also why are you setting variables if you are not using them. also let implies these variables will change.